dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaAPIHostsTask.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager;
+
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.MetadataDumperUsageException;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.ClouderaManagerHandle.ClouderaClusterDTO;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+
+/**
+ * The tasks dump data from the <a
+ * href="https://archive.cloudera.com/cm7/7.11.3.0/generic/jar/cm_api/apidocs/json_ApiHostList.html">Hosts
+ * API</a> which doesn't contain usage and disk data and collected as a fallback.
+ */
+public class ClouderaAPIHostsTask extends AbstractClouderaManagerTask {
```

Comment:
For this and other subclasses of `AbstractClouderaManagerTask`, inheritance needs to be addressed. The options are:
- design and document to support future inheritance
- prevent inheritance
(Effective Java 3rd ed, Item 19)

Both options are valid. Personally I'd use the second, because it seems faster to implement - all that is needed would be making the classes `final` and minor visibility changes (removing `protected`, because all subclasses are in the same package anyway).

Reply:
I'm not sure that this advice somehow improves the current design in the Dumper. Moreover this practice is not adopted, so this connector will look different compare to others. 
If you think that this point is essential, let's discuss it with the entire team and create a separate ticket to apply the practice if we will decide to adopt it.